src/diffpy/fourigui/fourigui.py

```diff
@@ -370,6 +370,10 @@ def intensity_upd_local(self):
             plane = self.cube[:, self.plane_num.get(), :]
         elif self.axis.get() == 2:
             plane = self.cube[:, :, self.plane_num.get()]
+
+        if np.all(np.isnan(plane)):
+            plane = np.zeros_like(plane)
+
```

Comment:
condition setting all `NaNs` to zero

Reply:
If we do that, that means we are simply converting `nan` values to all 0s. but only for [nan, nan, ... , nan] but not when [nan, 0.1, 0.2, nan, ...]

so doesn't feel like a good solution to me - when we call `np.nanmax`, I think it will pick up 0 instead of `nan` and this might not be good since your code has a section where it calculates `nan_ratio`.

I am not sure why we use `nan` in the first place. Why `nan` instead of `0`? Having `nan` around for an array where it should have the same type for the particular array normally, it feels quite odd.

Reply:
I think this is not the preferred behavior.  Setting NaN's to 0 falls under the umbrella of "magic".  Especially as scientists we want to avoid updating people's data.

Caden said "The goal of the test is to verify that the applycutoff method is working as expected for a given qmin and qmax." which of course is a tautology.  That is what a test does.  But the question is what do "we expect" that function to do.  In other words, what is our desired behavior?